### PR TITLE
[1.1.1] Fix PEM_write_bio_PrivateKey_traditional() to not output PKCS#8

### PR DESCRIPTION
**Pull request by levitte**
PEM_write_bio_PrivateKey_traditional() uses i2d_PrivateKey() to do the
actual encoding to DER.  However, i2d_PrivateKey() is a generic
function that will do what it can to produce output according to what
the associated EVP_PKEY_ASN1_METHOD offers.  If that method offers a
function 'old_priv_encode', which is expected to produce the
"traditional" encoded form, then i2d_PrivateKey() uses that.  If not,
i2d_PrivateKey() will go on and used more modern methods, which are
all expected to produce PKCS#8.

To ensure that PEM_write_bio_PrivateKey_traditional() never produces
more modern encoded forms, an extra check that 'old_priv_encode' is
non-NULL is added.  If it is NULL, an error is returned.

-----

This is a backport of #12728 to 1.1.1

### CODE REVIEW

**Review by t8m — APPROVED** [review]
LGTM

File: crypto/pem/pem_pkey.c [hunk 1]
```diff
@@ -108,6 +108,11 @@ int PEM_write_bio_PrivateKey_traditional(BIO *bp, EVP_PKEY *x,
                                          pem_password_cb *cb, void *u)
 {
     char pem_str[80];
+
+    if (x->ameth == NULL || x->ameth->old_priv_encode == NULL) {
+        ERR_raise(ERR_LIB_PEM, PEM_R_UNSUPPORTED_PUBLIC_KEY_TYPE);
```
Comment (t8m) [inline crypto/pem/pem_pkey.c:113]: Heh, should have looked at the Travis first - of course ERR_raise is 3.0 function.

**Review by t8m — CHANGES_REQUESTED** [review]
Needs replacement of ERR_raise

File: crypto/pem/pem_pkey.c (hunk 1 above)
Comment (levitte) [reply crypto/pem/pem_pkey.c:113]: Oh, of course. Will fix

Comment (levitte) [reply crypto/pem/pem_pkey.c]: Done

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
Merged

56456c3404b0ec27f93816d951ff7a58827481f0 Fix PEM_write_bio_PrivateKey_traditional() to not output PKCS#8

**Pull request closed without merging**